### PR TITLE
Gpu demo

### PR DESCRIPTION
**Pull request by QuietSwami**
PR to include GPU Demo

### CODE REVIEW

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 1]
```diff
@@ -45,25 +51,26 @@ services:
...
 ## Deployment
 
 In Nuvla.io, go to the App Store and find the app called **"GPU Demo”**. Click **“launch”**
 A panel with appear - select your NuvlaBox credentials. 
-As we don't have environmental variables or files, you can click **"Deploy"**
+As we don't have environmental variables or files, we can click **"Deploy"**
 
 At start, the app will download and build Darknet - this will take a while.
-Afterwards, it will launch the web server, so at this point you can access the web interface:
+Afterwards, it will launch the web server, so at this point we can access the web interface:
```
Comment (cjdcordeiro) [inline docs/tutorials/nuvlabox/gpu-demo.md]: add screenshot

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 2]
```diff
@@ -45,25 +51,26 @@ services:
...
     
     http://{IP of your machine}:5000
```
Comment (cjdcordeiro) [inline docs/tutorials/nuvlabox/gpu-demo.md:81]: add another screenshot

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 3]
```diff
@@ -12,30 +12,36 @@ has_children: false
...
+Follow the [NuvlaBox setup to connect](https://docs.nuvla.io/nuvlabox/nuvlabox-engine/quickstart.html) your device to Nuvla. 
 
-As described above, Docker-Compose has no command to use de GPU directly. 
-To use it we need to specify the run command in our Docker-Compose file. It looks like this:
+## Docker Compose with GPU
+
+To know which GPU to use, and its libraries, we use the GPU Peripheral Manager, which can be deployed with NuvlaBox. 
+The ouput is something similar to this: 
+
+![gpuDemo](/assets/peripheral-manager-gpu.png){: :center}
```
[the remark this answers is not in the document]
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md]: its a small mistake. I forgot to add the img folder to the path

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 4]
```diff
@@ -45,25 +51,26 @@ services:
         ports:
```
[the remark this answers is not in the document]
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md:54]: yes

**Comment by cjdcordeiro** [comment]
@QuietSwami when all the comments have been addressed please either comment here or re-request a review

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 5]
```diff
@@ -12,30 +12,36 @@ has_children: false
...
-use a docker-compose command to pass the GPU. 
+Since Docker 19, GPU usage when running a container can be achieved using "--gpus" all - this flag allows Docker to pass the GPU devices to the container.
+This capability is only available to Nvidia GPUS. While the Docker client is able to use the GPU flag, Docker Compose hasn't caught up yet. This means we can't natively use a Compose file to make use of the --gpus property
 
-We'll deploy a Docker container through Nuvla. To be able to use the GPU, we need to specify the devices and the libraries that are used by the GPU, and pass them through with Docker-Compose.
+Let's say we have a Compose-file defined application to be deployed from Nuvla. Given the aforementioned Docker limitation, we need to manually select the GPU devices and libraries from the host, and pass them to the container via the Compose file.
 
-To know the correct devices and libraires, you can use the GPU discovery tool installed with NuvlaBox.
+To know the correct devices and libraires, we can use the GPU discovery tool installed with NuvlaBox.
```
[the remark this answers is not in the document]
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md]: Which type of reference?

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 6]
```diff
@@ -12,30 +12,36 @@ has_children: false
...
```
[the remark this answers is not in the document]
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md:21]: what type of diagram?

[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/tutorials/nuvlabox/gpu-demo.md:21]: something like https://docs.nuvla.io/assets/img/videoAIDemo.png

something that visually shows the workflow of the demo

File: docs/tutorials/nuvlabox/gpu-demo.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/tutorials/nuvlabox/gpu-demo.md]: direct link to the code

[the remark this answers is not in the document]
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md]: To the github repo?

[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/tutorials/nuvlabox/gpu-demo.md]: y

**Comment by QuietSwami** [comment]
Everything should be done. Please let me know if any problems occur.

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 7]
```diff
@@ -1,73 +1,99 @@
...
+Since Docker 19, GPU usage when running a container can be achieved using `--gpus all` - this flag allows Docker to pass the GPU devices to the container.
+This capability is only available to NVIDIA GPUs. While the Docker client is able to use the GPU flag, Docker Compose hasn't caught up yet. This means we can't natively use a Compose file to make use of the `--gpus` property.
 
-We'll deploy a Docker container through Nuvla. To be able to use the GPU, we need to specify the devices and the libraries that are used by the GPU, and pass them through with Docker-Compose.
+Let's say we have a Compose file-defined application to be deployed from Nuvla. Given the aforementioned Docker limitation, we need to manually select the GPU devices and libraries from the host, and pass them to the container via the Compose file.
 
-To know the correct devices and libraires, you can use the GPU discovery tool installed with NuvlaBox.
+To know the correct devices and libraries, we can use the [GPU discovery tool](https://github.com/nuvlabox/peripheral-manager-gpu) installed with NuvlaBox.
+
+![gpuDemoSchema](/assets/img/gpu-demo-scheme.png){: :center}
```
Comment (cjdcordeiro) [inline docs/tutorials/nuvlabox/gpu-demo.md:22]: do you have the rights to use that image? Also, it shouldn't say "private app store". This is a tutorial so the app in Nuvla must be made public for the readers to try

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 8]
```diff
@@ -1,73 +1,99 @@
...
-Afterwards, it will launch the web server, so at this point you can access the web interface:
+Afterwards, it will launch a web server with the video feed, and the app's dynamic URL will be filled in Nuvla. So at this point we can access the web interface:
     
-    http://{IP of your machine}:5000
+    http://{IP of your machine}:5000    
+
+And the interface should look something like this:
+
+
+![gpuDemo](/assets/img/gpu-demo-interface.png){: :center}
```
Comment (cjdcordeiro) [inline docs/tutorials/nuvlabox/gpu-demo.md:94]: same question as above, about rights to use the image

File: docs/tutorials/nuvlabox/gpu-demo.md [hunk 9]
```diff
@@ -1,73 +1,99 @@
...
-        ports:
-            - "5000:5000"
-        devices:
-            - /dev/nvidiactl
-            - /dev/nvidia-uvm
-            - /dev/nvidia0
-        volumes:
-            - /usr/bin/:/usr/bin/
+  darknet:
+    image: 'franciscomendonca/darknet:1.0'
```
Comment (cjdcordeiro) [inline docs/tutorials/nuvlabox/gpu-demo.md:54]: pls push to sixsq's repo

File: docs/tutorials/nuvlabox/gpu-demo.md (hunk 7 above)
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md:22]: I made the image. Or are you talking about an asset in the image?

Comment (cjdcordeiro) [reply docs/tutorials/nuvlabox/gpu-demo.md:22]: the bicycle/car image

Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md:22]: No, that comes with Darknet. I'll change it

File: docs/tutorials/nuvlabox/gpu-demo.md (hunk 8 above)
Comment (QuietSwami) [reply docs/tutorials/nuvlabox/gpu-demo.md:94]: I'll change for the same image as above

File: docs/tutorials/nuvlabox/gpu-demo.md (hunk 7 above)
Comment (cjdcordeiro) [reply docs/tutorials/nuvlabox/gpu-demo.md:22]: after fixing the camera position and having a real stream, pls replace this image

**Pull request merged**